### PR TITLE
Nano: add `trainer.trace(accelerator="onnxruntime")`'s support

### PR DESCRIPTION
**Pull request by TheaperDeng**
This PR implement `trainer.trace` only, skip save/load for now waiting for #4487 .
`trainer.quantize`'s support for onnxruntime and the removal of old `eval_onnx()` API will be carried out in following PRs.

### CODE REVIEW

**Comment by TheaperDeng** [comment]
http://10.112.231.51:18888/job/BigDL-NanoTests-PR-Validation/28/
http://10.112.231.51:18888/job/BigDL-PRVN-chronos-Python-Spark-2.4-py37-ray-part1/736/

**Comment by TheaperDeng** [comment]
@zhentaocc maybe we just merge this PR and keep moving? Since there are no special internal/API design in this PR and most of the code follow what we did in openvino.

**Comment by zhentaocc** [comment]
LGTM.

File: python/nano/src/bigdl/nano/deps/onnxruntime/__init__.py [hunk 1]
```diff
@@ -13,3 +13,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+try:
+    import onnx
+    import onnxruntime as ort
+except ImportError:
+    raise ImportError("To enable onnxruntime inference, you need to install it by:\n"
+                      "\t\t pip install onnxruntime")
```
Comment (zhentaocc) [inline python/nano/src/bigdl/nano/deps/onnxruntime/__init__.py:21]: Is this right place to check?

Comment (TheaperDeng) [reply python/nano/src/bigdl/nano/deps/onnxruntime/__init__.py]: Good catch, I should put it to `onnxruntime/core`

**Comment by TheaperDeng** [comment]
http://10.112.231.51:18888/job/BigDL-NanoTests-PR-Validation/30/

**Comment by TheaperDeng** [comment]
http://10.112.231.51:18888/job/BigDL-NanoTests-PR-Validation/41/

**Pull request merged**